tests/CsrfTest.php

```diff
@@ -1,46 +1,68 @@
 <?php
-namespace Slim\HttpCache\Tests;
 
-use Psr\Http\Message\RequestInterface;
+declare(strict_types=1);
+
+namespace Slim\Tests\Csrf;
+
+use PHPUnit\Framework\TestCase;
 use Psr\Http\Message\ResponseInterface;
-use Slim\Http\Body;
-use Slim\Http\Collection;
-use Slim\Http\Environment;
-use Slim\Http\Headers;
-use Slim\Http\Request;
-use Slim\Http\Response;
-use Slim\Http\Uri;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
 use Slim\Csrf\Guard;
-
-class CsrfTest extends \PHPUnit_Framework_TestCase
+use Slim\MiddlewareDispatcher;
+use Slim\Psr7\Environment;
+use Slim\Psr7\Factory\ResponseFactory;
+use Slim\Psr7\Factory\UriFactory;
+use Slim\Psr7\Headers;
+use Slim\Psr7\Request;
+use Slim\Psr7\Response;
+use Slim\Psr7\Stream;
+
+class CsrfTest extends TestCase
 {
     /**
      * PSR7 request object
      *
-     * @var Psr\Http\Message\RequestInterface
+     * @var \Psr\Http\Message\RequestInterface
      */
     protected $request;
 
     /**
      * PSR7 response object
      *
-     * @var Psr\Http\Message\ResponseInterface
+     * @var \Psr\Http\Message\ResponseInterface
      */
     protected $response;
 
+    /**
+     * @var ResponseFactory
+     */
+    protected $responseFactory;
+
+    /**
+     * @var MiddlewareDispatcher
+     */
+    protected $middlewareDispatcher;
+
     /**
      * Run before each test
+     *
+     * @return void
      */
-    public function setUp()
+    public function setUp(): void
```

Comment:
As mentioned in my previous comment, we should use prophecies instead of using an actual implementation to test. I can rewrite the tests for this if you're not quite familiar with that.

Reply:
Actually, I'm learning test with that PR. I can try to test it myself I think 😄 .

Reply:
If you can, yes, I prefer if you rewrite the tests 😄.